pkg/ext-proc/scheduling/filter.go

```diff
@@ -0,0 +1,127 @@
+package scheduling
+
+import (
+	"fmt"
+	"math"
+
+	klog "k8s.io/klog/v2"
+
+	"ext-proc/backend"
+)
+
+type Filter interface {
+	Name() string
+	Filter(b *LLMRequest, pods []*backend.PodMetrics) ([]*backend.PodMetrics, error)
+}
+
+// filter applies current filterFunc, and then recursively applies next filters depending success or
+// failure of the current filterFunc.
+// It can be used to construct a flow chart algorithm.
+type filter struct {
```

Comment:
You might want to clarify that `nextOnSuccessOrFailure` is not considered in the following cases:
- When it's a success and `nextOnSuccess` is specified.
- When it's a failure and `nextOnFailure` is specified.

Reply:
Thanks for the suggestion. Added comment to clarify this.